Lombiq.Hosting.Tenants.Management/Controllers/ShellSettingsEditorController.cs

```diff
@@ -47,29 +49,56 @@ public async Task<IActionResult> Edit(ShellSettingsEditorViewModel model)
             return NotFound();
         }
 
-        var settingsDictionary = JsonConvert.DeserializeObject<Dictionary<string, string>>(model.Json);
-        var newSettings = new Dictionary<string, string>();
-
-        foreach (var key in settingsDictionary.Keys.Where(key => string.IsNullOrEmpty(settingsDictionary[key])))
+        model.Json ??= "{}";
+        if (!IsValidJson(model.Json))
         {
-            settingsDictionary[key] = null;
+            return RedirectToAction(
+                nameof(AdminController.Edit),
+                typeof(AdminController).ControllerName(),
+                new
+                {
+                    area = "OrchardCore.Tenants",
+                    id = model.TenantId,
+                });
         }
```

Comment:
Instead of clearing the user's input on a validation error, keep in the input and display an error message.

Reply:
I tried using the ModelState, and then the IUpdateModelAccessor. Even if I add a model error the error is not displayed.

Reply:
Please debug what the issue might be.

In any case, you can also add an error message via `INotifier`.

Reply:
> Please debug what the issue might be.

Did that before I put this to Done. So the controller action we are calling doesn't care about modelstate.
Adding INotifier.

Reply:
In the filter you can try to re-add the freshly saved content of the text field on validation error so it's not lost.

Reply:
So did you try re-adding the freshly saved content?

Reply:
Yeah I tried but didn't succeed. Then I guess I overrode this in my brain with the other comment's "The rest of the code needn't change." sentence.... Sorry, added it now.